File: backend/TaskApp.Functions/Models/Task.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TaskApp.Functions.Models
{
    [Table("Tasks")]
    public class TaskEntity
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; } = string.Empty;

        [MaxLength(1000)]
        public string? Description { get; set; }

        public bool IsCompleted { get; set; } = false;

        public int Priority { get; set; } = 2; // 1=High, 2=Medium, 3=Low

        public DateTime? DueDate { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? CompletedAt { get; set; }

        [Required]
        public Guid UserId { get; set; }

        [MaxLength(500)]
        public string? Tags { get; set; }

        public int? CategoryId { get; set; }
    }
}

File: frontend/src/types/Task.ts
export interface Task {
  id: number;
  title: string;
  description?: string;
  isCompleted: boolean;
  priority: number;
  dueDate?: string;
  createdAt: string;
  completedAt?: string;
  userId: string;
  tags?: string;
  categoryId?: number;
  categoryName?: string;
  categoryColor?: string;
}

export interface CreateTaskDto {
  title: string;
  description?: string;
  priority: number;
  dueDate?: string;
  tags?: string;
  categoryId?: number;
}

export interface UpdateTaskDto {
  title?: string;
  description?: string;
  isCompleted?: boolean;
  priority?: number;
  dueDate?: string;
  tags?: string;
  categoryId?: number;
}

// Backend returns capitalized properties
export interface ApiResponse<T> {
  Success: boolean;
  Message: string;
  Data?: T;
  Error?: string;
}
